js/markDone.js

```diff
@@ -0,0 +1,35 @@
+import { EDIT, COMPLETE } from "/js/constants.js";
+import calculateDuration from "/js/utility/calculateDuration.js";
+import createDurationElement from "/js/utility/createDurationElement.js";
+
+export default function markTodoAsCompleted(
+  todoItemElement,
+  taskList,
+  elementId
+) {
+  const task = taskList.find((item) => item.id === elementId);
+  if (!task) {
+    return;
+  }
+
+  task.done = true;
+
+  const textStyle = todoItemElement.querySelector(".text").style;
+  textStyle.textDecoration = "line-through";
+  textStyle.color = "green";
+
+  todoItemElement.querySelector(`[data-job="${COMPLETE}"]`).style.display = "none";
+  todoItemElement.querySelector(`[data-job="${EDIT}"]`).style.display = "none";
+
+  const taskStartDay = todoItemElement.querySelector(".date").innerText;
+  let taskStartDateString = taskStartDay.replace(/[^0-9]/g, "-");
+  taskStartDateString = taskStartDateString.slice(12);
+
+  const duration = calculateDuration(taskStartDateString);
```

Comment:
each task should keep a property `startDate`.

see https://github.com/ban-wtag/ToDoAppJS/pull/4#discussion_r1171095843

```suggestion
  const duration = calculateDuration(task.startDate);
```

Reply:
taskStarDateString = task.startDate; done!